docs/source/learn/eager.md

```diff
@@ -24,24 +24,40 @@ res = model(input)
 torch_xla.sync()
 ```
 
-The actual model compilation and device execution happens when
-`torch_xla.sync` is called. There are multiple drawback of this
-approach.
+While this approach enables performance optimizations, it introduces significant
+usability challenges.
 
-1.  Users are often confused about when the framework is tracing and
-    when the framework is executing.
-2.  Non-core model code(data preprocessing for example) often generates
-    some small pending execution that gets leaked into the main
-    graph(step function) and causes recompilation. The recompilation of
-    the whole graph is usually very expensive.
-3.  It is hard to debug when/why recompilation happens.
+## Challenges with LazyTensor Mode
 
-To mitigate above issues we want to introduce the new UX with eager and
-compile.
+- **Ambiguity**: Developers struggle to distinguish between tracing and
+execution phases, complicating development and debugging.
 
-## Basic Usage
+- **Recompilation Overhead**: Non-core operations (e.g., data preprocessing) can
```

Comment:
I think it could be clearer to explain that these non-core operations are all recorded. Maybe this phrasing:

```
- **Recompilation Overhead**: Whenever any part of the captured graph changes, `torch_xla.sync()` will
recompile the whole graph. Changes in non-core operations (e.g., data preprocessing) thus trigger expensive
recompilations.
```